src/AppSystem/Launcher.vala

```diff
@@ -176,6 +177,27 @@ public class Dock.Launcher : BaseItem {
         };
         bounce_up.done.connect (bounce_down.play);
 
+        var shake_animation_target = new Adw.CallbackAnimationTarget ((val) => {
+            var height = overlay.get_height ();
+            var width = overlay.get_width ();
+
+            overlay.allocate (
+                width, height, -1,
+                new Gsk.Transform ().translate (Graphene.Point () { x = (int) val })
+            );
+        });
+
+        shake = new Adw.TimedAnimation (
+            this,
+            0,
+            0,
+            70,
+            shake_animation_target
```

Comment:
Can you replace `shake_animation_target` with the `new Adw.CallbackAnimationTarget...` thing above it. We usually avoid using one-time variables